include/pybind11/common.h

```diff
@@ -321,6 +321,7 @@ struct internals {
     std::unordered_map<const void *, void*> registered_types_py;       // PyTypeObject* -> type_info
     std::unordered_multimap<const void *, void*> registered_instances; // void * -> PyObject*
     std::unordered_set<std::pair<const PyObject *, const char *>, overload_hash> inactive_overload_cache;
+    std::unordered_map<std::type_index, std::vector<bool (*)(PyObject *, void *&)>> direct_conversions;
```

Comment:
It wasn't clear to me, why this unordered map was needed, as compared to storing the ``std::vector<>`` (and not a pointer to it) directly in the ``py::detail::type_info``.